### PR TITLE
Fix NullPointerException cause game crashed

### LINKED ISSUE #14: Erro NullPointerException
Exit:

The game crashed whilst initializing game
Error: java.lang.NullPointerException: Cannot read the array length because "stitchResults" is null

Crash Log:
[crash-2024-12-08_12.10.19-client.txt](https://github.com/user-attachments/files/18052752/crash-2024-12-08_12.10.19-client.txt)

Minecraft: 1.20.1
Forge: 47.3.11
Mod version: 1.1.0

### PR DESCRIPTION
**Pull request by Wulian233**
### Fixes #14

Fix the NullPointerException error caused by `stitchResults` being null during game initialization.

* Add a null check for `stitchResults` before attempting to read its length in the `loadCachedData` method.
* Return an empty `ConcurrentHashMap` if `stitchResults` is null in the `loadCachedData` method.

### Changes

The mod description supports localization

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/steves-underwater-paradise/blinkload/pull/15?shareId=7b21f3c2-6c8c-478a-999f-ba14c676aadc).

---

Edit: Well, it seems like I fixed more than just a bug, I also added a new smalle feature and upgraded the mod version to 1.2

### CODE REVIEW

**Comment by Wulian233** [comment]
@Steveplays28 I tried to fix this crashing error with Copilot, could you review if this is the correct fix? Thank you! I don't know if the other two crashing errors are also caused by this

https://github.com/steves-underwater-paradise/blinkload/issues/13
https://github.com/steves-underwater-paradise/blinkload/issues/12

File: common/src/main/java/io/github/steveplays28/blinkload/client/cache/BlinkLoadCache.java [hunk 1]
```diff
@@ -105,6 +105,9 @@ public static void cacheData(@NotNull StitchResult stitchResult) {
 		try (@NotNull Reader reader = new FileReader(CACHED_DATA_FILE)) {
 			// Convert the JSON data to a Java object
 			@NotNull var stitchResults = JsonUtil.getGson().fromJson(reader, StitchResult[].class);
+			if (stitchResults == null) {
+				return new ConcurrentHashMap<>();
```
Comment (Wulian233) [inline common/src/main/java/io/github/steveplays28/blinkload/client/cache/BlinkLoadCache.java:102]: I'm not sure whether should `return cachedData` here. I also used Copilot on https://github.com/steves-underwater-paradise/blinkload/issues/11, and got the same fix as the issue above, but with a different return statement. Maybe we can solve all 4 issues at the same time.

https://github.com/steves-underwater-paradise/blinkload/issues/11

**Comment by Steveplays28** [comment]
Thanks @Wulian233, I'll do a review in the weekend. I don't have time currently

**Comment by Steveplays28** [comment]
@Wulian233 If you update Gradle to `v8.11.1` you also need to update Architectury Loom to `v1.9-SNAPSHOT`

**Comment by Wulian233** [comment]
Sorry I forgot. Upgrading many dependencies can always be less than pleasant🙂, I hope this won't break anything

**Comment by Steveplays28** [comment]
No worries, should be ok. I've tested that version combination already

**Comment by sherpascott** [comment]
I was having this same issue, but the issue stopped happening recently. I didn't update the mod or anything, so it must be a mod conflict. I'm now saving about 15 seconds on modpack loading times!

**Comment by Steveplays28** [comment]
@sherpascott I'm glad to hear that the mod helps!
I'll get this PR merged as soon as possible to fix all open issues on these crashes.

**Comment by Wulian233** [comment]
Note: Date `## 'v1.2.0' - 14/12/2024` in `changelog.md`
Need to change again at the time of merge (kind reminder PR review)

**Comment by Steveplays28** [comment]
I forgot to run the CI, seems like the wrapper validation is failing.

**Comment by Steveplays28** [comment]
I've fixed it in #17.

**Pull request merged**